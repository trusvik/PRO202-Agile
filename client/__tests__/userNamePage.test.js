// UserNamePage.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import UserNamePage from '../components/user/userNamePage';

// Mock the `useNavigate` hook
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => jest.fn(),
}));

describe('UserNamePage Snapshot Test', () => {
    it('renders correctly', () => {
        const setIsUserNameEntered = jest.fn(); // Mock function for setIsUserNameEntered

        // Render the component
        const tree = renderer.create(
            <MemoryRouter>
                <UserNamePage setIsUserNameEntered={setIsUserNameEntered} />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
