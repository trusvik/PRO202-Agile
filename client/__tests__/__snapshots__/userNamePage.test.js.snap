// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNamePage Snapshot Test renders correctly 1`] = `
<div
  className="mainBody"
>
  <div
    className="pageContainer"
  >
    <h1
      className="pageLogo"
    >
      Loading...
    </h1>
    <h2>
      Skriv inn navnet ditt
    </h2>
    <input
      className="inputBox"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Navn..."
      type="text"
      value=""
    />
    <button
      className="enterNameBtn"
      onClick={[Function]}
      type="submit"
    >
      OK
    </button>
  </div>
</div>
`;
